src/dev/mod.ts

```diff
@@ -33,18 +25,15 @@ interface Manifest {
   islands: string[];
 }
 
-export async function collect(directory: string): Promise<Manifest> {
+export async function collect(
+  directory: string,
+  islandsPaths: string[],
```

Comment:
Should probably be an optional `options` parameter, with sensible defaults:

```ts
const defaultOptions = {
  islandsPaths: ["./islands"],
};

type Options = Partial<typeof defaultOptions>;

export async function collect(
  directory: string,
  options?: Options
): Promise<Manifest> {
  options = {
	...defaultOptions,
	...options,
  };
  // ...
```

Reply:
Or maybe the option should be named `islandsPatterns` instead to allow for incorporating similar logic to https://github.com/denoland/fresh/pull/692, either within the same PR or as a future enhancement. So you could have something like this:

```ts
await dev(import.meta.url, "./main.ts", {
  islandsPatterns: [
    "./islands",
    "./**/*.island.tsx",
    "./**/*.island.jsx",
  ],
});
```

Reply:
Updated to have paths and globs as separate options since it's not trivial to differentiate.

Reply:
Couldn't matching be done with similar logic to `find <...> ! -type d`?

```sh
$ mkdir -p islands/a/b/c
$ touch islands/a/b/c/match.tsx match.island.tsx ignore-me.tsx
$ find ./islands ./**/*.island.tsx ! -type d

./islands/a/b/c/match.tsx
./match.island.tsx
```

This would mean the path `./islands` matches against subdirectories, too, which is a bit different from the current behavior. IMO allowing subdirectories is a more sensible default anyway, but if it's preferred not to do so for some reason, the default option could be switched to `["./islands/*.*"]` (assuming all relevant files will have an extension such as `.tsx`).

Concretely, that could be implemented something like this:

```js
import { join } from "https://deno.land/std@0.150.0/path/mod.ts"

const islandPaths = ["./islands", "./**/*.island.tsx"];

const globify = async (dirOrGlob: string) => {
  if (dirOrGlob.includes("*")) {
    return dirOrGlob;
  }

  try {
    const { isDirectory } = await Deno.stat(dirOrGlob);
    
    return isDirectory ? join(dirOrGlob, "**") : dirOrGlob;
  } catch {
    return dirOrGlob;
  }
};

for await (const glob of islandPaths.map(globify)) {
  for await (const entry of expandGlob(glob)) {
    if (entry.isFile) {
      // ...
    }
  }
}
```

Reply:
I don’t have too much of an opinion. I’ll leave it up to @lucacasonato to make the final call on how he’d like the options to work and update the PR with the decided on way. Would love to see this move forward to make sharing islands possible.